.azurepipelines/Platform-Build-Job.yml

```diff
@@ -19,120 +19,172 @@ parameters:
-        build_flags: $(Build.Flags)
-        run_flags: $(Run.Flags)
-        run_timeout: 25
-        extra_install_step: ${{ parameters.extra_install_step }}
-        install_tools: ${{ eq(parameters.container_image, '') }}
-        artifacts_identifier: '$(Build.Package) ${{ parameters.tool_chain_tag }} $(Build.Target) $(Build.ExtraTag)'
-        artifacts_binary: $(Build.ArtifactsBinary)
-        artifacts_other: $(Build.ArtifactsOther)
+  - ${{ each item in parameters.build_matrix }}:
+    - job: PlatformBuild_${{ item.Key }}
+      displayName: Build Platforms ${{ item.Key }}
+      condition: |
+        and(not(Canceled()),
+            or(${{ ne(item.Value.SelfHostAgent, true) }}, ne(variables['${{ parameters.pool_name }}'], '')))
+
+      workspace:
+        clean: all
+
+      variables:
+      - name: Run
+        ${{ if or(eq(item.Value.Run, true), eq(parameters.os_type, 'Linux')) }}:
+          value: true
+        ${{ else }}:
+          value: false
+
+      ${{ if and(ne(parameters.container_image, ''), ne(item.Value.SelfHostAgent, true)) }}:
+        container: ${{ parameters.container_image }}
+
+      ${{ if eq(item.Value.SelfHostAgent, true) }}:
+        pool:
+          name: $(${{ parameters.pool_name }})
+          demands:
+          - Agent.OS -equals ${{ parameters.os_type }}
+      ${{ else }}:
+        pool:
+          vmImage: ${{ parameters.vm_image }}
+
+      steps:
+      - ${{ if and(and(ne(parameters.container_image, ''), ne(item.Value.SelfHostAgent, true)), not(contains(parameters.vm_image, 'windows'))) }}:
+        - script: echo "##vso[task.prependpath]/home/vsts_azpcontainer/.local/bin"
+          displayName: Add User Local Bin to Path
+        - script: sudo dnf install -y mtools dosfstools
+          displayName: Install mtools and dosfstools
+
+      # Potential Extra steps from specific build targets
+      - ${{ item.Value.BuildExtraStep }}
+
+      - template: Steps/BuildPlatform.yml@mu_devops
+        parameters:
+          tool_chain_tag: $(tool_chain_tag)
+          build_pkg: ${{ item.Value.BuildPackage }}
+          build_target: ${{ item.Value.BuildTarget }}
+          build_arch: ${{ item.Value.BuildArch }}
+          build_file: ${{ item.Value.BuildFile }}
+          build_flags: ${{ item.Value.BuildFlags }}
+          run_flags: ${{ item.Value.RunFlags }}
+          run_timeout: ${{ parameters.run_timeout }}
+          extra_install_step: ${{ parameters.extra_install_step }}
+          install_tools: ${{ and(ne(item.Value.SelfHostAgent, true), eq(parameters.container_image, '')) }}
+          artifacts_identifier: '${{ item.Value.BuildPackage }} $(tool_chain_tag) ${{ item.Value.BuildTarget }} ${{ item.Value.BuildExtraTag }}'
+          artifacts_binary: ${{ item.Value.BuildArtifactsBinary }}
+          ${{ if or(eq(item.Value.Run, true), eq(parameters.os_type, 'Linux')) }}:
+            artifacts_other: ${{ item.Value.BuildArtifactsOther }}
+
+    - job: RunQemuBuilds_${{ item.Key }}

```

Comment:
Could you remind me why this has to be done in the repo rather then the original build platform template?

Reply:
Ignore me. This is to run the VS2022 build on Linux seperately since it can't be done in job, right? I think I misread this on the first pass

Reply:
Yeah, those were newly added here. Hopefully we can remove the extra run steps soon when QEMU is supported on Windows ARM64.